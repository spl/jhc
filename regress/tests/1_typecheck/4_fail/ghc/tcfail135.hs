-- A missing kind check made GHC 6.4 crash on this one

module ShoudlFail where

class Foo f where
	baa :: f a -> f

instance Foo Maybe where
	baa z = z
