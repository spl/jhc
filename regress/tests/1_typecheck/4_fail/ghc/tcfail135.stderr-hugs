ERROR "tcfail135.hs":6 - Illegal type "a b" in constructor application
