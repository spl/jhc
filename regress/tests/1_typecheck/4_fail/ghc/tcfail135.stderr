
tcfail135.hs:6:23:
    Expecting one more argument to `f'
    In the type `f a -> f'
    In the class declaration for `Foo'
